main.go

```diff
@@ -85,7 +86,8 @@ var (
 	vSphereClusterIdentityConcurrency int
 	vSphereDeploymentZoneConcurrency  int
 
-	tlsOptions = flags.TLSOptions{}
+	tlsOptions         = capiflags.TLSOptions{}
+	diagnosticsOptions = capiflags.DiagnosticsOptions{}
```

Comment:
looks like we don't use diagnosticsOptions to set options on the manger

Reply:
Should be set now if I'm understanding right - sorry I didn't pay any attention to getting the secure metrics serving feature into this PR - didn't even consider it :sweat_smile:

Reply:
Looks good, thx!